library/src/main/java/com/novoda/downloadmanager/WaitForLockRunnable.java

```diff
@@ -0,0 +1,44 @@
+package com.novoda.downloadmanager;
+
+import com.novoda.notils.logger.simple.Log;
+
+class WaitForLockRunnable {
+
+    interface Action {
+        void performAction();
+    }
+
+    static WaitForLockAndThenPerformActionRunnable waitFor(Object lock) {
```

Comment:
Introduced a nested builder so that we can do 

`WaitForLockRunnable.waitFor(downloadService).thenPerform(action)`